CriteoPublisherSdk/Tests/IntegrationTests/CR_BidManagerIntegrationTests.m

```diff
@@ -80,15 +78,13 @@ - (void)tearDown {
 }
 
 // Prefetch should populate cache with given ad units
-- (void)test_given3AdUnits_whenPrefetchBid_thenGet3Bids {
-  [self.bidManager
-      prefetchBidsForAdUnits:@[ self.cacheAdUnit1, self.cacheAdUnit2, self.cacheAdUnit3 ]
-                 withContext:self.contextData];
+- (void)test_given2AdUnits_whenPrefetchBid_thenGet2Bids {
+  [self.bidManager prefetchBidsForAdUnits:@[ self.cacheAdUnit1, self.cacheAdUnit2 ]
+                              withContext:self.contextData];
 
-  [self _waitNetworkCallForBids:@[ self.cacheAdUnit1, self.cacheAdUnit2, self.cacheAdUnit3 ]];
+  [self _waitNetworkCallForBids:@[ self.cacheAdUnit1, self.cacheAdUnit2 ]];
   XCTAssertNotNil([self.bidManager getBidThenFetch:self.cacheAdUnit1 withContext:self.contextData]);
   XCTAssertNotNil([self.bidManager getBidThenFetch:self.cacheAdUnit2 withContext:self.contextData]);
-  XCTAssertNotNil([self.bidManager getBidThenFetch:self.cacheAdUnit3 withContext:self.contextData]);
```

Comment:
Why do you need to remove such test ?

Reply:
Would have needed to mock static injection of `integrationRegistry` inside `AdUnitHelper` which should be easier to do once refactored on the other end. 

I did not want to make this PR last for too long, but if you consider the test relevant, I can work this out.

Reply:
The `AdUnitHelper` dependency is fetch statically, but still, it is fetch during the `bidManager getBidThenFetch` call. So you can mock the AdUnitHelper in the static dependency provider meantime.

I see that this test (https://github.com/criteo/ios-publisher-sdk/pull/234/files#diff-df5582f92dc8c24b9c4a3a4e05d75ddad728a4fa10babedeba1e89c3d7ae0f1aR200) is verifying that the feature don't work on InHouse as expected.

But I don't see any functional test checking that it is working for DFP. Do you mind adding such tests (like `DfpBannerFunctionalTests.m`) testing that, without mocks (except the network), it is possible to get a bid response for DFP integration (no need to go E2E with DFP ad server, but you're welcome if you want to).

Reply:
I've updated the content to allow these tests.

As we're relying on global shared object I need to set the value on this same static object, when `AdUnitHelper` refactor will be completed we can refactor this to benefit from IoC.

Also added a `CR_DfpRewardedFunctionalTests.m` file that does the opposite of the `InHouseFunctionalTest` checking that we're issuing http requests for Rewarded Ads.